contracts/Lotto.sol

```diff
@@ -0,0 +1,287 @@
+    // Max range for numbers (starting at 0)
+    uint8 internal maxValidRange_;
+
+    // Represents the status of the lottery
+    enum Status { 
+        NotStarted,     // The lottery has not started yet
+        Open,           // The lottery is open for ticket purchases 
+        Closed,         // The lottery is no longer open for ticket purchases
+        Completed       // The lottery has been closed and the numbers drawn
+    }
+    // All the needed info around a lottery
+    struct LottoInfo {
+        uint256 lotteryID;          // ID for lotto
+        Status lotteryStatus;       // Status for lotto
+        uint256 prizePoolInCake;    // The amount of cake for prize money
+        uint256 costPerTicket;      // Cost per ticket in $cake
+        uint8[] prizeDistribution;  // The distribution for prize money
+        uint256 startingBlock;      // Block timestamp for star of lotto
+        uint256 closingBlock;       // Block timestamp for end of entries
+        uint256 endBlock;           // Block timestamp for claiming winnings
+        uint8[] winningNumbers;     // The winning numbers
+    }
+    // Lottery ID's to info
+    mapping(uint256 => LottoInfo) internal allLotteries_;
+
+    //-------------------------------------------------------------------------
+    // EVENTS
+    //-------------------------------------------------------------------------
+
+    event newLotteryCreated(
+        uint256 indexed lottoID,
+        Status lotteryStatus,
+        uint8[] prizeDistribution,
+        uint256 prizePoolInCake,
+        uint256 costPerTicket,
+        uint256 startingBlock,
+        uint256 closingBlock,
+        uint256 endBlock,
+        address indexed creator
+    );
+
+    event newBatchMint(
+        address indexed minter,
+        uint256[] ticketIDs,
+        uint32[] numbers,
+        uint256 totalCost,
+        uint256 discount,
+        uint256 pricePaid
+    );
+
+    //-------------------------------------------------------------------------
+    // MODIFIERS
+    //-------------------------------------------------------------------------
+
+
+
+    //-------------------------------------------------------------------------
+    // CONSTRUCTOR
+    //-------------------------------------------------------------------------
+
+    constructor(
+        address _cake, 
+        uint8 _sizeOfLotteryNumbers,
+        uint8 _maxValidNumberRange    
+    ) {
+        cake_ = IERC20(_cake);
+        sizeOfLottery_ = _sizeOfLotteryNumbers;
+        maxValidRange_ = _maxValidNumberRange;
+    }
+
+    function init(address _lotteryNFT) public onlyOwner() {
+        nft_ = ILottoNFT(_lotteryNFT);
+    }
+
+    //-------------------------------------------------------------------------
+    // VIEW FUNCTIONS
+    //-------------------------------------------------------------------------
+
+    /**
+      * @return Gets the current time
+      */
+    function getTime() public view returns(uint256) {
```

Comment:
what's the purpose of this function? It can be obtained from the blockchain itself

Reply:
Had it there out of habit, other UI devs I've worked with like having a function for it. Will remove.